### DISCUSSION TITLE
Can we do interpolation string in slot ?

### DISCUSSION THREAD

**Discussion by jitendraP-ashutec** [lit/lit#3537] [Q&A]
Let say I have have one web component and it has slot with interpolation string.

for example  ` <web-com  name='name'">  hello ${name} </web-com>`  now in web-com I want to interpolation string what passed in slot, like we have in Angular. can we do such type of thing? does any body has idea about it ?

**Comment by usergenic** [comment]
Hi.  I am trying to understand the question correctly -- Are you asking whether the interpolation itself can be handled by the component that defines the slots?  Like, can the "web-com" element provide the value for the "name" identifier in the interpolation?  The short answer is that "name" will be resolved by the JavaScript language before the value of it is passed to the `html` tagged template literal function and as such the "web-com" element would not have an opportunity to provide a value using a lookup-by-name.

**Reply by jitendraP-ashutec** [reply]
@usergenic , Yes you understand question correctly. Thanks your answer, but Does any way to implement like this or you have any idea how we can do that?


Thanks
Jitendra Prajapati

**Reply by rictic** [reply]
You could pass a function to <web-com> from the parent, something like: https://lit.dev/playground/#gist=ae390db90fae11859c4426209074f0e2

We have more planning for a more detailed protocol going on at https://github.com/lit/lit/pull/3591
